tests/test_transformations/test_knn_clustering.py

```diff
@@ -0,0 +1,137 @@
+import unittest
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from sklearn.datasets import load_iris
+from spac.transformations import knn_clustering
+
+
+class TestKnnClustering(unittest.TestCase):
+    def setUp(self):
+        """
+        This method is run before each test.
+
+        It sets up a test AnnData object `syn_data` with the following attributes:
+
+        `syn_data.X`
```

Comment:
you don't actually use syn_data.X anywhere apart from this docstring. let's remove it from the docstring